cmd/bnsd/scenarios/escrow_test.go

```diff
@@ -0,0 +1,69 @@
+package scenarios
+
+import (
+	"testing"
+	"time"
+
+	"github.com/iov-one/weave/cmd/bnsd/app"
+	"github.com/iov-one/weave/cmd/bnsd/client"
+	"github.com/iov-one/weave/coin"
+	"github.com/iov-one/weave/x/escrow"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestQueryEscrowExists(t *testing.T) {
+	walletResp, err := bnsClient.GetWallet(escrowContract.Address())
+	// then
+	require.NoError(t, err)
+	require.NotNil(t, walletResp)
+	require.Len(t, walletResp.Wallet.Coins, 1)
+	assert.True(t, walletResp.Wallet.Coins[0].Whole > 0)
+}
+
+func TestEscrowRelease(t *testing.T) {
+	// query distribution accounts start balance
+	walletResp, err := bnsClient.GetWallet(distrContractAddr)
+	require.NoError(t, err)
+	startBalance := coin.Coin{Ticker: "IOV"}
+	if walletResp != nil {
+		startBalance = *walletResp.Wallet.Coins[0]
+	}
+	aNonce := client.NewNonce(bnsClient, alice.PublicKey().Address())
+	// when releasing 1 IOV by the arbiter
+	_, _, escrowID, _ := escrowContract.Parse()
+
+	addValidatorTX := &app.Tx{
+		Sum: &app.Tx_ReleaseEscrowMsg{
+			&escrow.ReleaseEscrowMsg{
+				EscrowId: escrowID,
+				Amount:   []*coin.Coin{{Ticker: "IOV", Whole: 1}},
+			},
+		},
+	}
+	_, _, contractID, _ := multiSigContract.Parse()
+	addValidatorTX.Multisig = [][]byte{contractID}
+
+	seq, err := aNonce.Next()
+	require.NoError(t, err)
+	require.NoError(t, client.SignTx(addValidatorTX, alice, chainID, seq))
+	resp := bnsClient.BroadcastTx(addValidatorTX)
```

Comment:
Doesn't BroadcastTx only return once the tx is in the block (commit succeeded)?

Reply:
ja, I have looked at the code in mempool.BroadcastTxCommit now and it does block or timeout. 👍

Reply:
We should comment the exact semantics in out client better.
I look at tendermint code too often to figure out timing issues